File: src/main/java/interview/greedy/MinimumAbsoluteDifferenceInAnArray/minimum_absolute_difference_in_an_array.py
#!/bin/python3

# import math
import os

# import random
# import re
# import sys


# Complete the minimumAbsoluteDifference function below.
def minimumAbsoluteDifference(arr):
    arr.sort()
    differences = [abs(arr[x + 1] - arr[x]) for x in range(0, len(arr) - 1)]
    return min(differences)


if __name__ == "__main__":
    fptr = open(os.environ["OUTPUT_PATH"], "w")

    n = int(input())

    arr = list(map(int, input().rstrip().split()))

    result = minimumAbsoluteDifference(arr)

    fptr.write(str(result) + "\n")

    fptr.close()

File: src/main/java/interview/greedy/MinimumAbsoluteDifferenceInAnArray/MinimumAbsoluteDifferenceInAnArray.cs
// <copyright file="MinimumAbsoluteDifferenceInAnArray.cs" company="PlaceholderCompany">
// Copyright (c) PlaceholderCompany. All rights reserved.
// </copyright>

namespace Interview.Greedy.MinimumAbsoluteDifferenceInAnArray
{
    using System;
    using System.CodeDom.Compiler;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Diagnostics.CodeAnalysis;
    using System.Globalization;
    using System.IO;
    using System.Linq;
    using System.Reflection;
    using System.Runtime.Serialization;
    using System.Text;
    using System.Text.RegularExpressions;

    public class MinimumAbsoluteDifferenceInAnArray
    {
        // Complete the minimumAbsoluteDifference function below.
        public static int MinimumAbsoluteDifference(int[] arr)
        {
            Array.Sort(arr);
            var differences = Enumerable.Range(0, arr.Length - 1)
            .ToList()
            .Select(index => Math.Abs(arr[index + 1] - arr[index]));
            return differences.Min();
        }

        private static void Main(string[] args)
        {
            var textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH") ?? string.Empty, true);

            var n = Convert.ToInt32(Console.ReadLine());

            int[] arr = Array.ConvertAll((Console.ReadLine() ?? string.Empty).Split(' '), arrTemp => Convert.ToInt32(arrTemp))
            ;
            var result = MinimumAbsoluteDifference(arr);

            textWriter.WriteLine(result);

            textWriter.Flush();
            textWriter.Close();
        }
    }
}